pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16381536?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16381536?buildTab=artifacts#/) on master @ [f334c73953bf5d5bed7a2c0effd72391284458f8](https://github.com/cockroachdb/cockroach/commits/f334c73953bf5d5bed7a2c0effd72391284458f8):


```
     true,
     true,
     true,
     true,
     true
    ]
   }
  ],
  "previousStatements": [
   "ALTER TABLE schema_w1_58.table_w1_69 ALTER PRIMARY KEY USING COLUMNS (\"col\"\"69_w1_72\", \"'col69_w1_73\", col69_w1_75)"
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: unimplemented: the following columns are not indexable due to their type: col"69_w1_72 (type refcursor), col69_w1_75 (type tsvector) (SQLSTATE 0A000)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3124
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: unimplemented: the following columns are not indexable due to their type: col"69_w1_72 (type refcursor), col69_w1_75 (type tsvector) (SQLSTATE 0A000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 96
Total Schema Statements Succeeded = 96
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 54
Total Transactions Rolled Back = 95
Total Transactions Executed = 149
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload3789403017
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload3789403017
--- FAIL: TestWorkload (13.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41107

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/5193e28c-e962-4ec8-8e27-bc43252bac18?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [2558f3f3a1dab0907db3dafd510ec5bc2464073f](https://github.com/cockroachdb/cockroach/commits/2558f3f3a1dab0907db3dafd510ec5bc2464073f):


```
   {
    "query": "SELECT EXISTS (SELECT * FROM public.table_w1_155)",
    "result": false
   },
   {
    "query": "WITH descriptors AS (SELECT crdb_internal.pb_to_json('cockroach.sql.sqlbase.Descriptor', descriptor)-\u003e'table' AS d FROM system.descriptor WHERE id = $1::REGCLASS) SELECT EXISTS (SELECT mut FROM (SELECT json_array_elements(d-\u003e'mutations') AS mut FROM descriptors) WHERE (mut-\u003e'primaryKeySwap') IS NOT NULL)",
    "queryArgs": [
     "public.table_w1_155"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w1_155 ADD COLUMN col_w1_155_w1_187 FLOAT4 NOT NULL"
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED COMMIT ERROR; Received an unexpected commit error: ERROR: transaction committed but schema change aborted with error: (23502): validation of column "col_w1_155_w1_187" NOT NULL failed on row: col155_w1_156='1979-11-11', col155_w1_157='17a5205f-70ee-43ac-9ad6-62d4df1cfefd', col155_w1_158=e'4\r D"', "col155%q_w1_159"='d2fd8938-979e-4e77-b316-140ccd3b4539', "co&l155_w1_160"=ARRAY['9942:7d7b:c6dd:bbd6:a873:d7b8:985:53d6/24','153.111.230.245/14','199.200.189.17','d9bb:b729:2143:3f33:a8b3:cbf7:7b05:e7a8/16'], "co%pl155_w1_161"=e'\r\x19\x14\x1aq', "col1\\U000D0BBE55_w1_162"=18, col̓155_w1_163='00000000-0000-0000-0000-000000000000', col_w1_155_w0_116=NULL, col_w1_155_w1_187=NULL (SQLSTATE 23502)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:659
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED COMMIT ERROR; Received an unexpected commit error
        	            	Wraps: (5) ERROR: transaction committed but schema change aborted with error: (23502): validation of column "col_w1_155_w1_187" NOT NULL failed on row: col155_w1_156='1979-11-11', col155_w1_157='17a5205f-70ee-43ac-9ad6-62d4df1cfefd', col155_w1_158=e'4\r D"', "col155%q_w1_159"='d2fd8938-979e-4e77-b316-140ccd3b4539', "co&l155_w1_160"=ARRAY['9942:7d7b:c6dd:bbd6:a873:d7b8:985:53d6/24','153.111.230.245/14','199.200.189.17','d9bb:b729:2143:3f33:a8b3:cbf7:7b05:e7a8/16'], "co%pl155_w1_161"=e'\r\x19\x14\x1aq', "col1\\U000D0BBE55_w1_162"=18, col̓155_w1_163='00000000-0000-0000-0000-000000000000', col_w1_155_w0_116=NULL, col_w1_155_w1_187=NULL (SQLSTATE 23502)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 97
Total Schema Statements Succeeded = 97
Total Schema Statement Expected Failures = 2
Total Transactions Committed = 61
Total Transactions Rolled Back = 91
Total Transactions Executed = 152
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_3_of_25/test.outputs/logTestWorkload3863471377
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload3863471377
--- FAIL: TestWorkload (27.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16402922?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16402922?buildTab=artifacts#/) on master @ [c64fb80786b82b0a848bf59e271d8b2ce44db674](https://github.com/cockroachdb/cockroach/commits/c64fb80786b82b0a848bf59e271d8b2ce44db674):


```
   {
    "query": "SELECT EXISTS (SELECT attrelid::REGCLASS AS table_name, attname AS column_name, pg_get_expr(adbin, adrelid) AS computed_formula FROM pg_attribute JOIN pg_attrdef ON (attrelid = adrelid) AND (attnum = adnum) WHERE (atthasdef AND (attrelid = $1::REGCLASS)) AND (pg_get_expr(adbin, adrelid) ILIKE (('%%' || $2) || '%%')))",
    "queryArgs": [
     "public.table_w1_4",
     "col1😽_w0_9"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w1_4 DROP COLUMN col1😽_w0_9"
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3139
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 10
Total Schema Statements Succeeded = 10
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 6
Total Transactions Rolled Back = 13
Total Transactions Executed = 19
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1905478014
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1905478014
--- FAIL: TestWorkload (4.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16403899?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16403899?buildTab=artifacts#/) on master @ [c64fb80786b82b0a848bf59e271d8b2ce44db674](https://github.com/cockroachdb/cockroach/commits/c64fb80786b82b0a848bf59e271d8b2ce44db674):


```
    "query": "SELECT EXISTS (SELECT attrelid::REGCLASS AS table_name, attname AS column_name, pg_get_expr(adbin, adrelid) AS computed_formula FROM pg_attribute JOIN pg_attrdef ON (attrelid = adrelid) AND (attnum = adnum) WHERE (atthasdef AND (attrelid = $1::REGCLASS)) AND (pg_get_expr(adbin, adrelid) ILIKE (('%%' || $2) || '%%')))",
    "queryArgs": [
     "schema_w0_5.table_w1_120",
     "col11%p5_w0_124"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE schema_w0_5.table_w1_120 DROP COLUMN \"col11%p5_w0_124\""
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3139
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 116
Total Schema Statements Succeeded = 116
Total Schema Statement Expected Failures = 2
Total Transactions Committed = 60
Total Transactions Rolled Back = 120
Total Transactions Executed = 180
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload3012296048
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload3012296048
--- FAIL: TestWorkload (14.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/a488de16-6191-4596-88c4-e8aa799c25ea?testReportRun=25&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [27468a22f82fb1e500bf7b35a62ca58d80abac39](https://github.com/cockroachdb/cockroach/commits/27468a22f82fb1e500bf7b35a62ca58d80abac39):


```
     "schema_w1_52.table_w1_155",
     "cOl172_w0_182"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE schema_w0_17.table_w0_145 RENAME COLUMN col145_w0_149 TO col_w0_145_w1_156",
   "ALTER TABLE schema_w0_17.table_w0_145 ADD CONSTRAINT \"table_w0_145_col145_w0_153_unique\" UNIQUE (col145_w0_153)",
   "ALTER TABLE schema_w1_52.table_w1_155 DROP COLUMN \"cOl172_w0_182\""
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: index "index_w1_155_w0_187" in the middle of being added, try again later (SQLSTATE 55000)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3124
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: index "index_w1_155_w0_187" in the middle of being added, try again later (SQLSTATE 55000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 119
Total Schema Statements Succeeded = 119
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 61
Total Transactions Rolled Back = 82
Total Transactions Executed = 143
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_25_of_25/test.outputs/logTestWorkload2579206003
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload2579206003
--- FAIL: TestWorkload (25.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=25</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fa9c845-a5d6-447e-9011-6fc56d2450f9?testReportRun=20&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [451cdbe687b3303665bfd490af59c5df777c8909](https://github.com/cockroachdb/cockroach/commits/451cdbe687b3303665bfd490af59c5df777c8909):


```
    "query": "SELECT $1:::REGCLASS = $2:::REGCLASS",
    "queryArgs": [
     "public.table_w1_2",
     "public.table_w1_2"
    ],
    "result": true
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w1_2 ADD CONSTRAINT table_w1_2_c😳olͪ2_w1_4_table_w1_2_c😳olͪ2_w1_4_fk FOREIGN KEY (c😳olͪ2_w1_4) REFERENCES public.table_w1_2 (c😳olͪ2_w1_4) ON DELETE CASCADE ON UPDATE CASCADE"
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: relation "table_w1_2" (112): unimplemented: cannot perform a schema change operation while a primary key change is in progress (SQLSTATE 0A000)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3124
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: relation "table_w1_2" (112): unimplemented: cannot perform a schema change operation while a primary key change is in progress (SQLSTATE 0A000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 13
Total Schema Statements Succeeded = 13
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 7
Total Transactions Rolled Back = 6
Total Transactions Executed = 13
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_20_of_25/test.outputs/logTestWorkload1105834529
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1105834529
--- FAIL: TestWorkload (7.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=20</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/877cba2b-c531-482b-813f-e87ab3415f97?testReportRun=11&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [2f0b0399a5c9656bdf8a417d0ab57e4eb1ae0f64](https://github.com/cockroachdb/cockroach/commits/2f0b0399a5c9656bdf8a417d0ab57e4eb1ae0f64):


```
    "queryArgs": [
     "schema_w0_29.table_w1_167",
     "schema_w0_29.table_w1_167"
    ],
    "result": true
   }
  ],
  "previousStatements": [
   "ALTER TABLE schema_w0_29.table_w1_167 ALTER PRIMARY KEY USING COLUMNS (c̎ol167_w1_174) USING HASH",
   "ALTER TABLE schema_w0_29.table_w1_167 ADD CONSTRAINT table_w1_167_col16😜7_w1_172_table_w1_167_col16😜7_w1_172_fk FOREIGN KEY (col16😜7_w1_172) REFERENCES schema_w0_29.table_w1_167 (col16😜7_w1_172) ON DELETE CASCADE ON UPDATE CASCADE"
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: relation "table_w1_167" (161): unimplemented: cannot perform other schema changes in the same transaction as a primary key change (SQLSTATE 0A000)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3124
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: relation "table_w1_167" (161): unimplemented: cannot perform other schema changes in the same transaction as a primary key change (SQLSTATE 0A000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 142
Total Schema Statements Succeeded = 142
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 74
Total Transactions Rolled Back = 118
Total Transactions Executed = 192
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_11_of_25/test.outputs/logTestWorkload3145812965
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload3145812965
--- FAIL: TestWorkload (27.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=11</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16433941?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16433941?buildTab=artifacts#/) on master @ [31aa9cf7c2844299f6447841e11e38ff4ed2ae8b](https://github.com/cockroachdb/cockroach/commits/31aa9cf7c2844299f6447841e11e38ff4ed2ae8b):


```
    "query": "SELECT EXISTS (SELECT attrelid::REGCLASS AS table_name, attname AS column_name, pg_get_expr(adbin, adrelid) AS computed_formula FROM pg_attribute JOIN pg_attrdef ON (attrelid = adrelid) AND (attnum = adnum) WHERE (atthasdef AND (attrelid = $1::REGCLASS)) AND (pg_get_expr(adbin, adrelid) ILIKE (('%%' || $2) || '%%')))",
    "queryArgs": [
     "schema_w0_118.table_w0_181",
     "\"col96_w1_101"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE schema_w0_118.table_w0_181 DROP COLUMN \"\"\"col96_w1_101\""
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3139
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:500
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:588
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 110
Total Schema Statements Succeeded = 110
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 56
Total Transactions Rolled Back = 111
Total Transactions Executed = 167
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload982728772
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload982728772
--- FAIL: TestWorkload (15.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128778 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.2 release-blocker]
- #128345 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/1c194dfd-48a0-4458-a824-09b2c9b6e215?testReportRun=24&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [abe3d27bee5120afa287df2f8cc81c82ab6864da](https://github.com/cockroachdb/cockroach/commits/abe3d27bee5120afa287df2f8cc81c82ab6864da):


```
    "query": "SELECT EXISTS (SELECT attrelid::REGCLASS AS table_name, attname AS column_name, pg_get_expr(adbin, adrelid) AS computed_formula FROM pg_attribute JOIN pg_attrdef ON (attrelid = adrelid) AND (attnum = adnum) WHERE (atthasdef AND (attrelid = $1::REGCLASS)) AND (pg_get_expr(adbin, adrelid) ILIKE (('%%' || $2) || '%%')))",
    "queryArgs": [
     "public.table_w1_50",
     "co'l50_w1_59"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w1_50 DROP COLUMN \"co'l50_w1_59\""
  ]
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: unimplemented: constraint "co'l50_w1_59_auto_not_null" in the middle of being added, try again later (SQLSTATE 0A000)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3133
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:499
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:587
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: unimplemented: constraint "co'l50_w1_59_auto_not_null" in the middle of being added, try again later (SQLSTATE 0A000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 51
Total Schema Statements Succeeded = 51
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 31
Total Transactions Rolled Back = 46
Total Transactions Executed = 77
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_24_of_25/test.outputs/logTestWorkload2436485193
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload2436485193
--- FAIL: TestWorkload (14.27s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=24</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128778 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/70ef7fd2-108e-4d7f-9014-33f091d294df?testReportRun=4&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [1993fc04b5116f20f4814d637c7ce87b003687e4](https://github.com/cockroachdb/cockroach/commits/1993fc04b5116f20f4814d637c7ce87b003687e4):


```
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:152
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).nextAdapter
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:246
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:152
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).next
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:250
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:282
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:320
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:928
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1990
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1704
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1707
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2427
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1974
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1181
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:146
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3436
        	            	  |   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:145
        	            	  | Wraps: (2) single entry value with no default column id
        	            	  | Error types: (1) *withstack.withStack (2) *errutil.leafError (SQLSTATE XX000)
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 148
Total Schema Statements Succeeded = 148
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 103
Total Transactions Rolled Back = 85
Total Transactions Executed = 188
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_4_of_25/test.outputs/logTestWorkload2455677245
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload2455677245
--- FAIL: TestWorkload (30.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128953 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.10-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16529222?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16529222?buildTab=artifacts#/) on master @ [33e2197dd224e01a46dd6c82e47ad54b36597872](https://github.com/cockroachdb/cockroach/commits/33e2197dd224e01a46dd6c82e47ad54b36597872):


```
    "queryArgs": [
     "udf_w1_36",
     "\"char\", BIT, BOOL, BOX2D, BYTES, CHAR, STRING, DATE, DECIMAL, FLOAT8, GEOGRAPHY, GEOMETRY, INET, INT8, INT2, INT2VECTOR, INT8, INTERVAL, JSONB, NAME, OIDVECTOR, PG_LSN, REFCURSOR, STRING, TIME, TIMESTAMP, TIMESTAMPTZ, TIMETZ, TSQUERY, TSVECTOR, \"char\", BIT, BOOL, BOX2D, BYTES, CHAR, STRING, DATE, DECIMAL, FLOAT8, GEOGRAPHY, GEOMETRY, INET, INT8, INT2, INT2VECTOR, INT8, INTERVAL, JSONB, NAME, OIDVECTOR, PG_LSN, REFCURSOR, STRING, TIME, TIMESTAMP, TIMESTAMPTZ, TIMETZ, TSQUERY, TSVECTOR, UUID, VARBIT, VARCHAR, VECTOR, VOID"
    ],
    "result": false
   }
  ],
  "previousStatements": [
   "CREATE FUNCTION public.udf_w1_36(val_0 \"char\", val_2 BIT, val_3 BOOL, val_4 BOX2D, val_5 BYTES, val_6 CHAR, val_7 STRING, val_8 DATE, val_9 DECIMAL, val_10 FLOAT8, val_11 GEOGRAPHY, val_12 GEOMETRY, val_13 INET, val_14 INT8, val_15 INT2, val_16 INT2VECTOR, val_17 INT8, val_18 INTERVAL, val_19 JSONB, val_20 NAME, val_22 OIDVECTOR, val_23 PG_LSN, val_24 REFCURSOR, val_30 STRING, val_31 TIME, val_32 TIMESTAMP, val_33 TIMESTAMPTZ, val_34 TIMETZ, val_35 TSQUERY, val_36 TSVECTOR, val_43 \"char\" DEFAULT e'\\x1a':::STRING, val_45 BIT DEFAULT B'011110110000011', val_46 BOOL DEFAULT false, val_47 BOX2D DEFAULT 'BOX(-0.3148512185410074 -0.5914869949795919,2.8571823374156886 -0.14053780780428482)':::BOX2D, val_48 BYTES DEFAULT NULL, val_49 CHAR DEFAULT 'w':::STRING, val_50 STRING DEFAULT e'\\U0001C2A2' COLLATE en, val_51 DATE DEFAULT '1994-02-04':::DATE, val_52 DECIMAL DEFAULT 4.657163749223521222E+20:::DECIMAL, val_53 FLOAT8 DEFAULT NULL, val_54 GEOGRAPHY DEFAULT '0103000060E6100000010000000C000000D59CCBB7B38565C0B4D6361BF3A53DC010B30A7F2767DA41B33FD8509E8B5FC090BF8BC3ECBD40C048A551E0109FE9C1F876FD974A8556C0CD35E4697A823AC0E44E7863B2AE004255082925F6E849C0F6E151C2AD0F1EC027CEA8ACB22DFDC16CCF5E40EDA7F73FBD93F615618314C05ED6C1491EB1F24182038F60B13E5A40C68A1F636334FC3F9110C19AE7A201C251BA7E9C83F06140D139D853EEC55340982C773F4F30CC410456DDF7E5BB40405A0B55BB8310564044E87054F85FFBC106DA21650A113DC097BCF9C0B4AA4440A13F0CDBB9BDFBC1BFD6F7AA93945FC07C25274996B84D403B8E8A517428E3C116AEEC19CF4761C0979E9C73C6F94B40A15AE0097588E3C1D59CCBB7B38565C0B4D6361BF3A53DC010B30A7F2767DA41':::GEOGRAPHY, val_55 GEOMETRY DEFAULT '0101000000A9210CE4B0C0F041D016E9989158FA41':::GEOMETRY, val_56 INET DEFAULT 'f59a:7fb4:2b84:ec4a:6e5:1ce0:7b4b:ec54/12':::INET, val_57 INT8 DEFAULT (-32768):::INT8, val_58 INT2 DEFAULT (-31292):::INT8, val_59 INT2VECTOR DEFAULT ARRAY[(-1957):::INT8, 9028:::INT8, 0:::INT8, (-3510):::INT8], val_60 INT8 DEFAULT 1495283582:::INT8, val_61 INTERVAL DEFAULT '1 day':::INTERVAL, val_62 JSONB DEFAULT '[[[{\"A4EsrGreg\": \"DUs\", \"ddgc4Ed\": null}, \"hlpiX\"], false, 1.4032355388950797, null], {\"0j\": null, \"b\": \"Dpq5Rz\"}, null, false]':::JSONB, val_63 NAME DEFAULT e'\\f\\x14\\x1b!\\x18':::STRING:::NAME, val_65 OIDVECTOR DEFAULT ARRAY[166571634:::OID, 0:::OID, 4243134178:::OID, 2097508820:::OID], val_66 PG_LSN DEFAULT '670D2DC5/487E9E21':::PG_LSN, val_67 REFCURSOR DEFAULT NULL, val_73 STRING DEFAULT e'+m!\\tF':::STRING, val_74 TIME DEFAULT '16:56:39.980023':::TIME, val_75 TIMESTAMP DEFAULT '1983-12-13 11:46:14.00078':::TIMESTAMP, val_76 TIMESTAMPTZ DEFAULT '2000-02-27 06:56:52.000207+00':::TIMESTAMPTZ, val_77 TIMETZ DEFAULT NULL, val_78 TSQUERY DEFAULT e'\\'WetC\\' \u003c-\u003e \\'oEZ\\' | \\'KfnPnEX\\'':::TSQUERY, val_79 TSVECTOR DEFAULT e'\\'NSjQ\\' \\'RNbHk\\' \\'fTXfrdhRAv\\' \\'vHSGlXhQv\\' \\'xBpyayi\\':715B':::TSVECTOR, val_81 UUID DEFAULT '36fcee26-1829-496b-b3b4-9aa1c61b828c':::UUID, val_82 VARBIT DEFAULT B'11011111011000010100001110111110101000001111001000011', val_83 VARCHAR DEFAULT e'/\\r$\u0026\\x17':::STRING, val_84 VECTOR DEFAULT '[9.737226e-08,5.6912214e+31,-1.0071023e+13,-6.9309537e+16,-7.717748e-24,-2.9181821e+20,4.8163363e+21,-1.0332957e-18,-5.196947e-16,8.586172e+30,-3.440719e-24,1.034925e-25,-2.7127844e-15,-1.05476545e+14,1.0688736e-34,-1.8674098e+37,4.299773e+36,3.9642756e+21,6.3953215e+28,1.3570907e+12,0.009368465,1.4930018e+18,-1.4841237e-22,2.9518474e+36,-18.435915,-5.423375e+25,-1.2240331e-11,-9.72633e-34,4.2391877e+21,6.7393937,4.8021852e-26,-1.9943055e+18,-8.174009e-25,-6.4816113e+09,1.1458555e-32,-2.924566e-36,4.4425587e-22,-1.8215823e-37,5.71954e-14,1.700117e-20,-6.16205e-15,-9.1872075e-20,8.3137356e-38,4.4054072e-14,-5.1854584e-38,1.3710566e+38,0.11959295,3.703383e+30,-5.8932343e-15,5.400298e-08,-8.321183e+34,-3.2693793e-27,2.5849175e-27,3.8004007e-19,-4.9532405e-28,-1.729055e-19,-1.9420832e+31,-1.2058734e+24,0.10447437,-1.0837623e+13,3.4023687e+31,-9.931226e+33,-7.534803e+13,-8.516568e+30,5.9129954e-28,1.6958498e+11,0.31080133,-8.8586745e-07,-1.5240635e+20,7.921152e+37,-6.9190875e-29,-1.9733501e-38,-1.14376e+17,-1.5689533e+29,118.171646,-0.00091552094,3.4030338e+22,-1.3872663e-36,2.5331967,0.0053073056,5.2946965e+22,347968.4,4.6091652e-38,-6.139822e+33,-2.1872233e-34,7.2133064e+07,-1.7718111e+22,-4.2273183e+33,-2.6927502e-37,-2.1877665e+19,-9.514401e+27,5.311869e+21,-2.2742137e+35,5.2951417e+20,5.3984753e+23,6.5386063e-10,2.7296118e-10,-0.005007083,9.6801126e+08,-8.692476e-05,-5.850072e+18,-3.898854e+17,2.3730424e+17,-1.0316808e-34,-0.00016438178,1.787445e-26,-739402.2,6.5921507e+37,8.455869e+22,1.01374526e+37,-1.7705929e+12,2.579968e+25,-3.2337863e+34,-2.4865197e-29,-1.5255858e+19,4.7804954e-14,3.2883128e-19,4.5576863e+30,-3.5517907e+26,4.8476924e-08,3.606874e-32,-4.859823e-28,-2.4153761e-24,-6.849528e+16,-1.8009953e+21,-5.977476e+17,-2.2416626e-07,1.6947568e+22,0.68519163,1.6413531e-26,7.8589474e+18,1.739973e-39,-1.9910207e-35,4.663519e+07,6.5731866e-36,2.3939774e-16,0.00032091592,1.0133928e+29,4.6353312e-23,-2.0400202e+19,1.1642175e-31,-5.449775e-11,6.749002e-12,-1.2045932e+23,-8.788308e-36,2.2297583e+38,-1.9795198e-07,-5.6848363e-12,6.592764e-05,-8.041477e-08,7.561031e+20,-3.710847,7.4837015e-25,4.806058e-05,1.9521519e-17,2.6548759e-12,6.207454e+33,-1.1467144e-33,4.1658788e-16,-2.6003859e+22,3.9950137e+27,6.7379876e+35,-8.3282714e-26,3.1280373e-19,-1.4266942e+26,2.9685145e-08,7.640799e-14,-3.221125e+36,1.0443439e+26,-1.5120904e-35,-1.6598179e-36,-0.14394526,-1.3525659e+09,2.1897271e+21,5.591006e-16,3.2800798e-36,2.2131164e-12,1.1930049e+30,-1.06371655e-10,0.10655899,217374.72,-1.3497584e-27,5.721933e+12,3.016833e+25,-400550.6,-1.1654629e-20,1.415555e-12,2.7962072e+34,1.3229699e-34,9.7734255e+20,-1.1294848e-35,146.13646,-4.112402e+09,-2.1921644e+35,-3.009465e+19,2.798017e-11,3.3680425e+13,4.1836203e+14,-1.0246894e-34,-1.3692966e+29,-3.757353e+08,-0.002622524,-2.5188007e-14,2.2843986e+32,2.6927267e-09,-6396.382,-1.4292894e+27,-6.2099285e+29,-2.1960938,-2.4725125e+10,93.60725,6.755952e-38,4.432654e-24,-9.1856e-20,7.962709e+10,-1.908722e+14,8.95553e-18,350.4415,-1.5036315e-32,-9.393103e-21,2.906288e-37,2.367477e-22,-2.6443692e+28,1.973122e+36,-1.3557411e+26,-20089.193,9.73049e+34,-4.5003492e-23,0.000112465845,3.6715432e-22,-1.0021735e-08,356728.75,5.854587e+11,1.9776802e+06,4.7924865e+33,-3.4220784e+17,-2.9475447e-24,-8.799243e+31,8.6770197e-32,-1.0730029e+07,5.65068e+20,-1.5956564e+33,-7.357154e-29,-1.0602513e+09,-1.7198392e-14,18514.65,4.2828288e+08,9.307592,1.2219516e-29,8.0224575e-37,1.7534658e+26,-1.597841e-10,-2.7117944e+29,14099.718,2.1565214e-32,-6.6229504e-06,2.459308e-09,-2.2134977e+21,5.428584e-34,-2.0820983e-09,6.1783347e-30,-7.2133296e+32,3.8336994e+31,-4.5157528e+10,-7.1446664e-26,-2.0482638e+11,-2.0959828e+38,-1.0344471e-31,-2.7456164e+10,-1.1144305e-36,-5.919096e-32,-1.3354552e+13,0.0020641559,52872.113,-0.02384032,-3.9029568e+09,-3.4785722e-14,-1.2907975e-31,-2.9524616e-08,3.1062286e-29,-4.448067e-28,-7.1346526e+17,-2.1071829e-22,1.3343291e-21,7.356126e-24,-6.197593e+09,-5.637799e+33,0.0046951715,6.319848e+36,-2.2732664e+22,-3.4353942e+17,2.6031975e-20,-0.08925522,2.3265246e-17,2.2605365e+06,3.8974957e+22,-2.433886e-06,7.166388e+21,-2.8984221e-27,5.139375e+34,-7.996361e+28,4.939649e+18,-3.809308e+22,0.0126155,-0.09488543,1.372972e-19,4.2080811e+18,-1.2712895e+29,-1.1401253e+14,-9.0311985e-18,-5.315931e+31,-5.5017773e-19,-7.813576e-11,3.8976916e-19,3.9244506e+35,-1.9338523e+25,-1.1817309e-34,-5.6246424e+12,-5.600489e+12,6.5302715e-33,-1.8916983e-16,1.4894215e+13,7.287807e-20,2.6391725e+24,4.4963926e+30,1.206446e+20,-1.1271752e+27,-1.5154707e-27,1.3051007e+09,4.663754e+17,5.217026e+15,-2.3758563e+38,-1.7189496e-13,-1.2112321e+36,6.637376e-28,1.5679283e-17,-2.6636492e+07,4.664004e-12,-5.3368753e-17,1.3778192e+15,-0.02617499,0.0020995922,2.372022e+36,-3.97153e-30,6689.702,-6.1549783e-25,-2.1733419e+26,3.479472e-17,6.912148e-06,-1.416787e+09,2.7078915e-21,0.000409569,-3.2273018e+13,2.4470066e-17,2.2619948e-34,-1.337829e-05,-3.9575885e-24,-1.5454194e+30,-1.9506231e+06,-8.013056e+14,-1.6216374e+33,1.2012894e+08,-3.9992368e+06,-1.2237544e-38,9.2260076e+17,-1.0294386e+24,1.1730839e-29,3.5345683e+36,3.8594532e-16,-1174.864,-6.9513965e-20,465169.06,1.1154316e-07,3.0436048e-10,-1961.8285,1.8398124e-27,4.698868e-35,8.847378e+31,4.9273037e-29,-2.637265e-13,-5.056881e+19,3.5466313e-12,8.396302e-23,4.5932028e+14,1.104659e-16,4.2103504e-16,4.9573524e-22,6.814948e-23,-2.3515963e+29,-6.968007e-08,99540.5,8.094474e-30,-9.2405946e-30,-2.2054202e+19,1.1078592e+15,-0.2374426,-7.604255e-18,2.7172566e+25,-6.2275024e-18,3.662461e-25,-1.0887708e+14,3.5516086e+19,-7.4072394e-11,3.4129495e+18,-5.9653336e-09,1.8240755e-31,-9.9556654e-14,7.833196e-33,-1674.7518,1.0786892e-20,1.6792483e+14,7.923919e+23,1.5217939e-25,-7.5611035e+34,-2.9542104e+32,-18.02766,-1.3391767,3.377056e-23,-1.154226e-11,7.823609e+26,-3.780322e-22,1.1727763e+24,-2.5068926e-19,-6.2229377e-09,0.7084055,0.07265428,-6.850451e-14,3.5457256e-06,4.5795146e+37,-7.377392e+23,-163.40808,-1.9631198e-11,-1.9830935e+34,1.204625e+34,29.884565,-2.5403626e+32,-4.5598306e-37,-3.6633702e+09,1.4483125e+19,-6.671361e-30,-6.8037e-36,1.8996206e-23,7157.988,5.3358445e+08,1.9421838e-10,-1.1412969e+34,1.3495755e+27,-1.2658705e-36,-2.5481345e+11,-3.50933e-31,3.4905154e-20,4.535427e-37,-3.027165e+36,2.2011734e+31,-74606.63,-549375.25,-3.933201e+13,-3.878672e+29,-4.3415223e-32,-2.706037e+29,-7.8393327e+31,1.1845051e-15,2.135627e-20,-8.936504e-05,-1.6325208e-28,-4.9595183e-06,-1.6501502e-10,7.778706e+19,-5.472599e-25,70873.82,-1.732437e+10,4.0509605e-07,-6.4319357e+10,1.0998317e+09,2.15187e+38,-0.0035514133,-6.8633726e+25,5.8183702e+26,3.1934672e+14,-0.00033962802,-3.3828304e+08,1.6674091e+07,2.8491418e-10,-7.145929e-26,4.0095048e+22,4.217666e-38,-4.3438456e+24,-9.4854904e+29,1.4154825e+16,-7.194689e+17,-4.2586716e-14,-7.2672335e+33,-4.0700366e-23,-3.7408837e+23,6.3475825e-10,9.0407426e-17,2.1492524e-30,7.2886415e+19,4.240598e+34,-1.3156544e+31,-3.3095307e+14,-5.0464486e+13,-9.758576e-17,1.8580922e+16,2.1275358e-29,-3.1183137e-19,3.4968646e-24,-8.794352e+31,1.1295136e+23,-2.7951079e-36,1.9776841e+28,-6.731679e-26,2.4252192e-36,-5.3210527e+24,-3.098432e+22,1.8335544e-29,4.4630066e-10,5.7052377e-13,4.8421585e+33,-2.8969916e-18,-1.8246979e+37,1.2707628e-22,-5.69353e+21,-1.5781503e-37,-1.0665319e-21,-6.2707504e+07,5.064891e+13,-8.715889e-37,1.2318633e+14,-9.639733e-25,-8.0126977e-16,8.896752e-19,-2.8529234e+22,2.581553e-23,-13171.912,-4.1912902e+08,-1.2555024e-07,-1.3442823e-24,0.0014117365,3.167485e-22,3.1006807e+31,5.6317947e+25,-1.4228549e+16,1.3056424e-31,365734.03,-5.1349394e+26,-3.6462045e+06,1.0120842e+06,2.6023258e+33,1.3102387e+25,1.2789903e+11,-4.1505282e+22,-4.2420446e+20,-2.484088e-36,8.9294374e+08,1.6235891e+32,-138.40227,8.991648e+13,-8.6574867e-16,-422.87546,162.95354,1.1083266e-06,0.7517536,-1.288083e+07,1.2287435e+27]':::VECTOR, val_85 VOID DEFAULT '':::VOID)\n\tRETURNS REFCURSOR\n\tLANGUAGE SQL\n\tAS $$ SELECT NULL WHERE TRUE $$"
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:134: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:134
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: argument of DEFAULT must be type void, not type string (SQLSTATE 42804)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3139
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:499
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:587
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:123
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: argument of DEFAULT must be type void, not type string (SQLSTATE 42804)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 20
Total Schema Statements Succeeded = 20
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 14
Total Transactions Rolled Back = 11
Total Transactions Executed = 25
    schema_change_external_test.go:99: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2716082611
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2716082611
--- FAIL: TestWorkload (5.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128953 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.10-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/udf_params
+++ b/pkg/sql/logictest/testdata/logic_test/udf_params
@@ -15,6 +15,9 @@ CREATE FUNCTION f(OUT param INT) RETURNS VOID AS $$ SELECT 1; $$ LANGUAGE SQL;
 statement error pgcode 42P13 pq: function result type must be int because of OUT parameters
 CREATE FUNCTION f(OUT param INT) RETURNS RECORD AS $$ SELECT 1; $$ LANGUAGE SQL;
 
+statement error pgcode 42P13 SQL functions cannot have arguments of type VOID
+CREATE FUNCTION f(param VOID) RETURNS UUID LANGUAGE SQL AS $$ SELECT NULL $$;
+
 statement ok
 CREATE FUNCTION f(OUT param INT) RETURNS INT AS $$ SELECT 1; $$ LANGUAGE SQL;
 

--- a/pkg/sql/opt/optbuilder/create_function.go
+++ b/pkg/sql/opt/optbuilder/create_function.go
@@ -192,8 +192,13 @@ func (b *Builder) buildCreateFunction(cf *tree.CreateRoutine, inScope *scope) (o
 		if param.Class == tree.RoutineParamInOut && param.Name == "" {
 			panic(unimplemented.NewWithIssue(121251, "unnamed INOUT parameters are not yet supported"))
 		}
-		if param.IsInParam() && typ.IsPolymorphicType() {
-			sawPolymorphicInParam = true
+		if param.IsInParam() {
+			if typ.Family() == types.VoidFamily {
+				panic(pgerror.Newf(pgcode.InvalidFunctionDefinition, "SQL functions cannot have arguments of type VOID"))
+			}
+			if typ.IsPolymorphicType() {
+				sawPolymorphicInParam = true
+			}
 		}
 		if param.IsOutParam() {
 			outParamTypes = append(outParamTypes, typ)

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -4066,7 +4066,8 @@ FROM
 		if typeVal.Identical(types.AnyTuple) ||
 			typeVal.IsWildcardType() ||
 			typeVal == types.RegClass ||
-			typeVal.Family() == types.OidFamily {
+			typeVal.Family() == types.OidFamily ||
+			typeVal.Family() == types.VoidFamily {
 			continue
 		}
 		if pgVectorNotSupported && typeVal.Family() == types.PGVectorFamily {

